### PR TITLE
add d.ts for pumpify

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/pumpify/pumpify-tests.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as pumpify from 'pumpify';
```
Comment (RyanCavanaugh) [inline types/pumpify/pumpify-tests.ts]: Classes and functions cannot be imported using `import * as` syntax. Please fix either the test or the definition file to use a correct import form.

See also https://stackoverflow.com/questions/39415661/what-does-resolves-to-a-non-module-entity-and-cannot-be-imported-using-this

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Comment by JustinBeckwith** [comment]
Thanks!  I fixed this up, and enabled `allowSyntheticDefaultImports` on the tsconfig.  Hope that's cool :)

**Comment by aoberoi** [comment]
@JustinBeckwith i'm having an issue with these definitions, thought i might run it by you in case you had any ideas. basically, the constructor doesn't take any arguments in this .d.ts, while in the actual package it does. how would you use the package without getting the error `Expected 0 arguments, but got N.`?

**Comment by aoberoi** [comment]
also, how do these labels work? i see `revision needed` but no comments (other than my own) since the PR was merged. does that label not get removed?

**Pull request merged**